cliquet/resource.py

```diff
@@ -120,30 +100,36 @@ def fetch_record(self):
                 content_type='application/json')
             raise response
 
-    def process_record(self, new, old=None):
-        """Hook to post-process records and introduce specific logics
-        or validation.
-        """
+    def _process_record(self, new, old=None):
         new = self.preprocess_record(new, old)
         return new
 
     def preprocess_record(self, new, old=None):
+        """Hook to pre-process records and introduce specific logics
+        or validation.
+
+        :param new: the validated record to be created or updated.
+        :type new: dict
+        :param old: the old record to be updated,
+            ``None`` for creation endpoints.
+        :type old: dict
+        """
         return new
 
-    def merge_fields(self, record, changes):
+    def _merge_fields(self, record, changes):
         """Merge changes into current record fields.
         """
         for field, value in changes.items():
             has_changed = record.get(field, value) != value
             if self.mapping.is_readonly(field) and has_changed:
                 error = 'Cannot modify {0}'.format(field)
-                self.raise_invalid(name=field, description=error)
+                self._raise_invalid(name=field, description=error)
 
         updated = record.copy()
         updated.update(**changes)
-        return self.validate(updated)
+        return self._validate(updated)
 
-    def validate(self, record):
+    def _validate(self, record):
```

Comment:
Should this stay public, in case a library user would like to define its own validation logic?